Field.h

```diff
@@ -125,4 +126,38 @@ class TextField: public Field {
         string getValue() {
             return fieldValue;
         }
+};
+
+class CharField: public Field {
+    
+    private:
+        string fieldValue;
+        int maxLength;
+        
+    public:
+        CharField(string name, int maxLen, string defaultValue="", bool primary=false, bool unique=false, bool nullable=true):
+            Field(name, CHAR, primary, unique, nullable, maxLen) 
+        { 
+            fieldValue = defaultValue;
+            maxLength = maxLen;
+        } 
+        
+        void setValue(string value) {
+            setNull(false);
+            fieldValue = checkMaxLen(value, maxLength);
+        }
+        
+        string getValue() {
+            return fieldValue;
+        }
+        int getMaxLen() {
+            return maxLength;
+        }
+        
+        string checkMaxLen(string newVal, int MaxLenVal) {
+            if (newVal.length() > MaxLenVal) {
+                throw length_error("Field value exeeds Max Length");
+            }
+            return newVal;            
+        }
 };
```

Comment:
PS I know it was already missing the newline, but just add it in while you are at it. There are a few other files like this as well.